### PR TITLE
Adds new method to contruct cubes

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

Mostly @paarongiroux's and @kaitlyndlee's work. I just fixed some syntax errors at the end. 

## How Has This Been Tested?

Tests seem to be passing on Jenkins 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/Cube/Cube.cpp [hunk 1]
```diff
@@ -73,6 +73,33 @@ namespace Isis {
     open(fileName.toString(), access);
   }
 
+  void Cube::fromLabel(const FileName &fileName, Pvl &label, QString access) {
```
Comment (krlberry) [inline isis/src/base/objs/Cube/Cube.cpp:85]: Needs docstring.

File: isis/src/base/objs/Cube/Cube.cpp [hunk 2]
```diff
@@ -73,6 +73,33 @@ namespace Isis {
...
+
+    for (auto grpIt = cubeLabel.beginGroup(); grpIt!= cubeLabel.endGroup(); grpIt++) {
+      putGroup(*grpIt);
+    }
+
+    close();
+    open(fileName.toString(), access);
+  }
+
+  void Cube::fromIsd(const FileName &fileName, Pvl &label, nlohmann::json &isd, QString access) {
```
Comment (krlberry) [inline isis/src/base/objs/Cube/Cube.cpp:114]: Needs documentation

**Review by krlberry — CHANGES_REQUESTED** [review]
:+1: This looks good - just needs documentation added for the two new methods added to the `Cube` class! Also, it looks like there are Jenkins failures for this? (Unless I'm looking at older failures.)

**Comment by Kelvinrr** [comment]
tests should be passing now but let it run overnight to be sure. @kberryUSGS If it passes and you re-review and find it approvable we can merge I think.

**Review by krlberry — APPROVED** [review]
:+1:

**Pull request merged**